common/wallet_tx.c

```diff
@@ -39,6 +46,37 @@ struct command_result *param_wtx(struct command *cmd,
         return NULL;
 }
 
+struct command_result *param_wtx_utxos(struct command *cmd,
+				 const char *name,
+				 const char *buffer,
+				 const jsmntok_t *tok,
+				 struct wallet_tx *wtx)
+{
+	size_t i;
+	const jsmntok_t *curr;
+	const char **outpoints = tal_arr(cmd, const char*, 0);
+
+	json_for_each_arr(i, curr, tok)
+		tal_arr_expand(&outpoints, json_strdup(tmpctx, buffer, curr));
+	if (!tal_count(outpoints))
+		return command_fail(cmd, JSONRPC2_INVALID_PARAMS,
+			"Could not decode any of the outpoints. The utxos should be"
+			" specified as an array of 'txid:output_index'.");
+
+	wtx->utxos = wallet_select_specific(cmd, cmd->ld->wallet, outpoints);
+
+	if (!wtx->utxos)
+		return command_fail(cmd, JSONRPC2_INVALID_PARAMS,
+			"Could not decode all of the outpoints. The utxos should be"
+			" specified as an array of 'txid:output_index'.");
+	if (tal_count(wtx->utxos) == 0)
+		return command_fail(cmd, JSONRPC2_INVALID_PARAMS,
+				"No matching utxo was found from the wallet."
+				"You can get a list of the wallet utxos with"
+				" the `listfunds` RPC call.");
```

Comment:
Parsing here, before asking wallet_select_specific(), makes this code path simpler too.  If you can't decode txid or outnum, you give a meaningful error here.

Just hand wallet_select_specific an array of txids and an array of outnums: a failure means some outpoint wasn't available.